include/k4arecord/playback.hpp

```diff
@@ -14,11 +14,218 @@
 #include <chrono>
 #include <string>
 #include <vector>
+#include <memory>
 
 namespace k4a
 {
 
-/** \class playback playback.hpp
+/** \class data_block playback.hpp <k4arecord/playback.hpp>
+ * Wrapper for \ref k4a_playback_data_block_t
+ *
+ * \sa k4a_playback_data_block_t
+ */
+class data_block
+{
+private:
+    class data_block_ref
+    {
+    public:
+        data_block_ref(k4a_playback_data_block_t handle) : m_handle(handle){};
+        ~data_block_ref()
+        {
+            reset();
+        }
+
+        void reset()
+        {
+            if (m_handle != nullptr)
+            {
+                k4a_playback_data_block_release(m_handle);
+                m_handle = nullptr;
+            }
+        }
+
+        k4a_playback_data_block_t get_handle()
+        {
+            if (m_handle != nullptr)
+            {
+                return m_handle;
+            }
+            return nullptr;
+        }
+
+        /** Returns true if two data_block_ref's refer to the same k4a_playback_data_block_t, false otherwise
+         */
+        bool operator==(const data_block_ref &other) const noexcept
+        {
+            return m_handle == other.m_handle;
+        }
+
+        /** Returns false if the data_block_ref is valid, true otherwise
+         */
+        bool operator==(std::nullptr_t) const noexcept
+        {
+            return m_handle == nullptr;
+        }
+
+        /** Returns true if two data_block_ref wrap different k4a_playback_data_block_t instances, false otherwise
+         */
+        bool operator!=(const data_block_ref &other) const noexcept
+        {
+            return m_handle != other.m_handle;
+        }
+
+        /** Returns true if the k4a_playback_data_block_t is valid, false otherwise
+         */
+        bool operator!=(std::nullptr_t) const noexcept
+        {
+            return m_handle != nullptr;
+        }
+
+    private:
+        k4a_playback_data_block_t m_handle;
+    };
+
+public:
+    /** Creates a data_block from a k4a_playback_data_block_t
+     * Takes ownership of the handle, you should not call k4a_playback_data_block_release? on the handle after
+     * giving it to the data_block; the data_block will take care of that.
+     */
+    data_block(k4a_playback_data_block_t handle = nullptr)
+    {
+        m_handle = std::make_shared<data_block_ref>(handle);
+        if (m_handle == nullptr)
```

Comment:
>if (m_handle == nullptr) [](start = 8, length = 24)

Pretty sure this can't ever happen - the only way for this to happen is if data_block_ref's constructor throws (in which case we won't get this far because a k4a::error was thrown) or if operator new throws (in which case we won't get this far because a std::bad_alloc has been thrown)